lib/libutils/ext/qsort_helpers.c

```diff
@@ -0,0 +1,189 @@
+// SPDX-License-Identifier: BSD-3-Clause
+/*
+ * Copyright (c) 2024, STMicroelectronics
+ */
+
+#include <stdlib.h>
+#include <util.h>
+
+static int qsort_cmp_int(const void *a, const void *b)
```

Comment:
There's a lot of repeated code here. How about something like:
``` C
#define QSORT_HELPLER(name, type)				\
static int cmp_ ## name(const void *a, const void *b)		\
{								\
	const type *ia = a;					\
	const type *ib = b;					\
								\
	return CMP_TRILEAN(*ia, *ib);				\
}								\
								\
void qsort_ ## name(type *aa, size_t n)				\
{								\
	qsort(aa, n, sizeof(*aa), cmp_ ## name);			\
}

QSORT_HELPLER(int, int);
QSORT_HELPLER(uint, unsigned int);
```

Reply:
Indeed, it would be more concise. I'll update. Thanks.